File: my-folder/0496-next-greater-element-i/solution.java
import java.util.Hashtable;

class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
       Hashtable<Integer, Integer> great = new Hashtable<Integer, Integer>();
		for (int i = 0; i < nums2.length; i++) {
			boolean found = false;
            System.out.println(nums2[i]);
			for (int j = i + 1; j < nums2.length; j++) {
				if (nums2[j] > nums2[i]) {
					great.put(nums2[i], nums2[j]);
					found = true;
					break;
				}
			}
			if (!found) {
				great.put(nums2[i], -1);
			}
		}
		int[] ans = new int[nums1.length];
		for (int i = 0; i < nums1.length; i++) {
			ans[i] = great.get(nums1[i]);
		}
		return ans;
    }
}

File: my-folder/0496-next-greater-element-i/solution.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int n2 = nums2.size();
        stack<int> s;
        vector<int> right(n2, -1);
        unordered_map<int, int> map2;
        for (int i = 0; i < nums2.size(); i++)
        {
            map2[nums2[i]] = i;
            while(!s.empty() && nums2[i] > nums2[s.top()])
            {
                right[s.top()] = nums2[i];
                s.pop();
            }
            s.push(i);
        }
        for (int i = 0; i < nums1.size(); i++)
        {
            nums1[i] = right[map2[nums1[i]]];
        }
        return nums1;
    }
};
